### PR TITLE
Remove some remnants of Python2

### PR DESCRIPTION
**Pull request by Lorak-mmk**
Support for Python < 3.7 was dropped some time ago, but there were still a lot of places in the codebase with workarounds required for those versions - most notably usage of six package.
This PR removes all usages of six, removes six from dependencies, removes all workarounds for older versions that I found and fixes comments / docs references.
There are 2 commits. First one drops six - it is pretty big, but the changes are mostly very simple and mechanical.
Second one removes other mentions / workaround, unrelated to six.

This PR is a based on https://github.com/scylladb/python-driver/pull/243 - just rebased it on this repository's master and solved the conflicts.

### CODE REVIEW

**Comment by fruch** [comment]
@Lorak-mmk 

https://github.com/datastax/python-driver/pull/1168 overlaps a bit of this.

Notice that PR updated the CI to use python 3.7, and it's not clear if the python2 support is removed on this repo..

Since the CI still runs python2

**Comment by Lorak-mmk** [comment]
Oh, I didn't see #1168 . My PR goes further anyway, so I don't think that much of a problem.

Do you know if this appveyor CI is really used here? I see it didn't pass even once for the past 2 years, always fails (I can't see older runs).

**Comment by absurdfarce** [comment]
Thanks for the PR @Lorak-mmk!

Have you signed the Contributor License Agreement for contributions to DataStax open source projects? If not you can find it at https://cla.datastax.com/. Thanks!

Regarding your question about CI: the AppVeyor configuration is out-of-date and can be ignored.  What's relevant for purposes of getting any changes merged is the output of the Travis test runs.

**Comment by Lorak-mmk** [comment]
I'm contributing this code on behalf of my employer ScyllaDB. We have signed Contributor License Agreement in the past when we were contributing fixes to Java driver. For more context see https://github.com/datastax/java-driver/pull/1195

**Comment by bschoening** [comment]
#1168 ran successfully with appveyor by removing python 2.7 from the yaml config.  You could cherry pick that change and/or add it to 1172.

With the impending freeze of Cassandra 5.x trunk, it would be great to get an updated Python driver w/out 'six' incorporated there.

**Comment by Lorak-mmk** [comment]
I cherry picked this commit, is it okay with you @vgali7 ?

**Comment by vgali7** [comment]
@Lorak-mmk Yeah certainly

**Comment by fruch** [comment]
@absurdfarce @bschoening, is there anything else blocking this one from getting merged ?

**Comment by bschoening** [comment]
+1

**Comment by absurdfarce** [comment]
Thanks for the question @fruch !  The only remaining issue with this PR that I'm aware of is that we need to complete a review of it.  That work is in process but I'm trying to juggle a lot of priorities at the moment so I haven't had a chance to dig into this in detail yet.

I can say that this change (or, more broadly, removing six from the code base) is one of the things we'd like to include in our next release (likely 3.29.0).  So it's very definitely on my radar... it's just a matter of finding time to get it reviewed and merged.

**Comment by bschoening** [comment]
Perhaps we could break this down into bit-sized chunks which can be quickly reviewed.

First, fixing AppVeyor by removing python 2.7 from the yaml config is really useful because it lets builds complete successfully. I could submit a separate PR for just that change.

Second, Vineet's #1168 was designed to be _quick_ to review as it *ONLY* removes 'if six.PY2' logic (and includes AppVeyor above)

All the above changes to completely remove six are great, but all together a bit complex to review.

File: cassandra/datastax/graph/graphson.py
Comment (absurdfarce) [inline cassandra/datastax/graph/graphson.py:138]: This should be the following instead:

```
if type(value) == int and (value > MAX_INT32 or value < MIN_INT32):
```

six.integer_types is just "int" in the Python3 case (it is a seq for Python2) so I'm not sure how this wasn't a problem before.

File: tests/integration/advanced/graph/fluent/test_graph.py
Comment (absurdfarce) [inline tests/integration/advanced/graph/fluent/test_graph.py:124]: keys() on a dict returns a dictionary view object rather than a sequence so this needs to be explicitly converted to an iterator.  So this should instead be:

```
property_name = next(iter(vertex_label.non_pk_properties.keys()))
```

File: tests/integration/advanced/graph/fluent/__init__.py
Comment (absurdfarce) [inline tests/integration/advanced/graph/fluent/__init__.py:462]: Another case where we need an iter:

```
property_name = next(iter(vertex_label.non_pk_properties.keys()))
```

File: tests/integration/advanced/graph/fluent/__init__.py
Comment (absurdfarce) [inline tests/integration/advanced/graph/fluent/__init__.py:540]: Another case where we need an iter:

```
property_name = next(iter(vertex_label.non_pk_properties.keys()))
```

File: tests/integration/advanced/graph/test_graph_datatype.py
Comment (absurdfarce) [inline tests/integration/advanced/graph/test_graph_datatype.py:90]: Another case where we need an iter:

```
property_name = next(iter(vertex_label.non_pk_properties.keys()))
```

File: tests/integration/advanced/graph/test_graph_datatype.py
Comment (absurdfarce) [inline tests/integration/advanced/graph/test_graph_datatype.py:171]: Another case where we need an iter:

```
property_name = next(iter(vertex_label.non_pk_properties.keys()))
```

File: tests/integration/advanced/graph/test_graph_query.py
Comment (absurdfarce) [inline tests/integration/advanced/graph/test_graph_query.py:590]: Another case where we need an iter:

```
prop_name = next(iter(vl.non_pk_properties.keys()))
```

**Review by absurdfarce — CHANGES_REQUESTED** [review]
Still working on a full review but I wanted to get info about these changes out as quickly as possible.

The canonical test setup for the Python driver is our Jenkins server which runs a full unit and integration test suite against a set of Apache Cassandra and DSE versions (neither Travis nor AppVeyor run the full integration test suite).  I setup a build of these changes on that server and observed several test failures due to a few nits that come along with the removal of six.  The comments in this review fixed all of the new issues introduced by this PR, so my expectation is that incorporating the changes referenced in this review will go a long way towards getting this in.

@Lorak-mmk and/or @fruch : if you guys are able to include these changes in this PR (and possibly in your downstream fork) that would be great.  I'm also fine including them in a follow-on commit after we get the work in this PR merged.  Just let me know what you want to do.

**Comment by Lorak-mmk** [comment]
I fixed the issues you pointed out, sorry it took so long. We didn't notice them before because we don't run DSE integration tests

**Comment by absurdfarce** [comment]
No worries @Lorak-mmk , thanks for the quick turnaround!  The rest of this looks pretty good so I think we're largely ready to merge this guy (and finally be completely rid of six!)

**Comment by fruch** [comment]
> No worries @Lorak-mmk , thanks for the quick turnaround!  The rest of this looks pretty good so I think we're largely ready to merge this guy (and finally be completely rid of six!)

@absurdfarce 

You might want take a look at also on:

https://github.com/scylladb/python-driver/pull/260


It's paving most of the way for python 3.12 support

**Comment by absurdfarce** [comment]
Thanks for the pointer @fruch!  I'll add a comment on the ongoing work on Python 3.12 support on our side there.

**Pull request merged**
